src/app/server/Mdns.cpp

```diff
@@ -144,6 +151,76 @@ CHIP_ERROR AdvertiseCommisioning()
     }
     advertiseParameters.SetShortDiscriminator(static_cast<uint8_t>(value & 0xFF)).SetLongDiscrimininator(value);
 
+    if (DeviceLayer::ConfigurationMgr().GetDeviceType(value) != CHIP_NO_ERROR)
+    {
+        ChipLogProgress(Discovery, "DNS-SD Device Type not set");
+    }
+    else
+    {
+        advertiseParameters.SetDeviceType(chip::Optional<uint16_t>::Value(value));
+    }
+
+    char deviceName[129];
+    if (DeviceLayer::ConfigurationMgr().GetDeviceName(deviceName, sizeof(deviceName)) != CHIP_NO_ERROR)
+    {
+        ChipLogProgress(Discovery, "DNS-SD Device Name not set");
+    }
+    else
+    {
+        advertiseParameters.SetDeviceName(chip::Optional<const char *>::Value(deviceName));
+    }
+
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
+    char rotatingDeviceIdHexBuffer[RotatingDeviceId::kHexMaxLength];
+    size_t rotatingDeviceIdHexBufferSize = 0;
+    ReturnErrorOnFailure(GenerateRotatingDeviceId(rotatingDeviceIdHexBuffer, rotatingDeviceIdHexBufferSize));
+    advertiseParameters.SetRotatingId(chip::Optional<const char *>::Value(rotatingDeviceIdHexBuffer));
+#endif
+
+    // TODO: check if commissioned yet, and if commissioned, use secondary
+    if (notYetCommissioned)
+    {
+        if (DeviceLayer::ConfigurationMgr().GetInitialPairingHint(value) != CHIP_NO_ERROR)
+        {
+            ChipLogProgress(Discovery, "DNS-SD Pairing Hint not set");
+        }
+        else
+        {
+            advertiseParameters.SetPairingHint(chip::Optional<uint16_t>::Value(value));
+        }
+
+        char pairingInst[129];
```

Comment:
add a constant for the 129 here and for the string sizes above.

This is probably not technically correct:
  - this stack variable has scope in the if branch only. advertiserParameter lifetime is larger than this, so you may not reference if from the optional that gets retained.